docs/proposals/compute-resource-metrics-api.md

```diff
@@ -0,0 +1,179 @@
+# Kubernetes compute resource metrics API
+
+## Goals
+
+Provide resource usage metrics on pods and nodes on the API server to be used
+by the scheduler to improve job placement, utilization, etc. and by end users
+to understand the resource utilization of their jobs.
+
+## Current state
+
+Right now, the Kubelet exports container metrics via an API endpoint. This
+information is not gathered nor served by the Kubernetes API server.
+
+## Use cases
+
+The first user will be kubectl. The resource usage data can be shown to the
+user via a periodically refreshing interface similar to `top` on Unix-like
+systems. This info could let users assign resource limits more efficiently.
+
+```
+$ kubectl top kubernetes-minion-abcd
+Pod: monitoring-heapster-abcde
+CPU:
+    mean: 0.12 cores
+    95th: 1.10 cores
+    max:  1.91 cores
+Memory:
+    mean: 300MB (10%)
+    95th: 330MB (11%)
+    max:  395MB (13%)
+```
+
+A second user will be the scheduler. To assign pods to nodes efficiently, the
+scheduler needs to know the current free resources on each node.
+
+## Proposed endpoints
+
+	/api/v1/namespaces/myns/podMetrics/mypod/[derived|raw]
+	/api/v1/nodeMetrics/myNode/[derived|raw] 
```

Comment:
Thought about derived/raw - If we go with the proposed `/api/v1/namespaces/myns/podMetrics/mypod/[derived|raw]`, I don't see how a user could get all the derived metrics for all the pods in a namespace.

If we split up derived/raw in different endpoints, like `/api/v1/namespaces/myns/podMetrics/[derived|raw]/mypod`, then you could do `/api/v1/namespaces/myns/podMetrics/[derived|raw]` to fetch a map (or list) containing all the metrics.